applications/segyio-catb.c

```diff
@@ -210,20 +149,41 @@ int main( int argc, char** argv ){
 
         if( err ) return errmsg(opterr, "Unable to read binary header");
 
-        for( int c = 0; c < 30; ++c ){
-            int field = get_binary_value( binheader, bfield_value[ c ] );
-            if( opts.nonzero && !field) continue;
-
-            if( opts.description ) {
-                int byte_offset = (bfield_value[ c ] - SEGY_TEXT_HEADER_SIZE);
-                printf( "%s\t%d\t%d\t%s\n",
-                    su[ c ],
-                    field,
-                    byte_offset,
-                    su_desc[ c ] );
+        int nr_fields = sizeof(field_data)/sizeof(binary_field_type);
+        for( int c = 0; c < nr_fields; ++c ){
+            segy_field_data fd = segy_get_field( binheader, field_data[c].offset );
+            if ( fd.error ) return errmsg( fd.error, "Unable to read field" );
+
+            if( opts.nonzero && fd.value.u32 == 0) continue;
+
+            int byte_offset = (field_data[c].offset - SEGY_TEXT_HEADER_SIZE);
+            const char* short_name = field_data[c].short_name;
+            const char* description = field_data[c].description;
+
+            if( fd.datatype == SEGY_UNSIGNED_INTEGER_8_BYTE ) {
+                long long int field = fd.value.u64;
+                if( opts.description )
```

Comment:
Preference note, disregard it if you want :cat: :
I often get burned by two-lines ifs and fors when they don't have {}. I overlook them not having {}, add some debug statement "inside" and then for the next 20 minutes wonder WTF is going on... :pouting_cat: 

So I would really appreciate adding braces to this if/else :sun_with_face: